src/hooks/useLocaleDirection.ts

```diff
@@ -9,11 +9,15 @@ import { isLangRightToLeft } from "@/lib/utils/translations"
  * Custom hook that sets the DOM direction based on the locale,
  * responding to changes in the locale without requiring refresh.
  */
-export const useLocaleDirection = (): void => {
+export const useLocaleDirection = (): string => {
+  const [direction, setDirection] = useState<"ltr" | "rtl">("ltr")
   const { locale } = useRouter()
 
   useEffect(() => {
     const dir = isLangRightToLeft(locale as Lang) ? "rtl" : "ltr"
     document.documentElement.setAttribute("dir", dir)
```

Comment:
@pettinarip Fyi, despite passing the `direction` to the Chakra theme provider, I'm still requiring `document.documentElement.setAttribute("dir", dir)` to automatically flip the direction when the locale is changed. Without this, switching from English to Farsi would require a refresh to update the DOM direction:

https://github.com/ethereum/ethereum-org-website/assets/54227730/88e1e662-b8d4-4ab2-8ded-9242da513a31

I too would prefer to remove the use of `document` here, but going to keep it for now until we find a solution that automatically updates this on locale change without a refresh.

Reply:
Got it! no worries, thanks for investigating.